File: internal/infra/di/webhook_orchestrator_mp.go
package di

import infra_aws "github.com/fiap/ms-payment/internal/infra/aws"

type WebhookOrchestratorMPContainer struct {
	LambdaClient *infra_aws.LambdaClient
}

func NewWebhookOrchestratorMPContainer(region infra_aws.AwsRegion, stage infra_aws.AwsStage) *WebhookOrchestratorMPContainer {
	lambdaClient := infra_aws.NewLambdaClient(region, stage)
	return &WebhookOrchestratorMPContainer{
		LambdaClient: lambdaClient,
	}
}

File: src/infra/di/webhook_orchestrator_mp.ts
import { LambdaAdapter } from "@driven_lambda/lambda"
import { LambdaClientWrapper } from "@aws/lambda_client"
import { AwsRegion, AwsStage } from "@aws/utils"

export class WebhookOrchestratorMPContainerFactory {
	lambdaAdapter: LambdaAdapter

	constructor(region: AwsRegion, stage: AwsStage) {
		const client = new LambdaClientWrapper(region, stage)

		this.lambdaAdapter = new LambdaAdapter(client)
	}
}